content/learn/book/welcome/plugins/_index.md

```diff
@@ -1,16 +1,96 @@
 +++
 title = "Modularity through plugins"
-weight = 4
+weight = 3
 template = "book-section.html"
 page_template = "book-section.html"
 +++
 
-TODO: Explain how plugins work
+In Bevy, virtually all of the functionality you might want is added via **plugins**, which are added to your app.
+These might provide basic windowing functionality, handle input or sound, calculate physics or provide game-specific logic like a tile map system.
 
-## `MinimalPlugins`
+By combining Bevy's first-party plugins with third-party alternatives, you can modularly customize the behaviour of the Bevy game engine, and then add your own plugins to contain your own game-specific code in a well-organized way.
+
+## Writing your own plugins
+
+Plugins are collections of code that modify the `AppBuilder` (which controls all of the data and logic of our game) using the builder pattern.
+Any code in a plugin could be directly substituted directly on the base `AppBuilder`.
+There's no magic to be found here; they're just a straightforward tool for code organization.
+
+You can write your own to organize your own code by implementing the `Plugin` trait on a struct of your own creation.
+
+```rust
+use bevy::prelude::*;
+
+fn main(){
+ App::build()
+  // As discussed below, DefaultPlugins provide the standard scaffolding for Bevy games
+   .add_plugins(DefaultPlugins)
+   // Plugins merely organize code: this could just be replaced directly with 
+   // the init_resource and add_system calls from below
+   .add_plugin(ScorePlugin)
+   .run();
+}
+
+struct ScorePlugin;
+
+impl Plugin for ScorePlugin {
+  fn build(&self, app: &mut AppBuilder) {
+     app
+       // The Score struct is addded as a resource (global singleton) to the world, 
+       // beginning at the default value of 0
+       .init_resource::<Score>()
+       // Increments the score by 1 every pass of the game loop
+       .add_system(increment_score.system())
+       // Prints the current value of the score
+       .add_system(report_score.system());
+  }
+}
+
+#[derive(Default, Debug)]
+struct Score(u8);
+
+fn increment_score(score: ResMut<Score>){
+  score.0 += 1;
+}
+
+fn report_score(score: Res<Score>){
+  dbg!(score);
+}
+```
 
 ## `DefaultPlugins`
```

Comment:
Code blocks in headings currently show up "too small". We'll want to either remove the code block or improve the formatting before publishing this.

Reply:
I'd prefer to fix the formatting: it's nice to clearly signify that it's a proper programming noun.

Reply:
Would it also benefit to link to the respective doc if it's a proper noun in bevy context?

Reply:
Link added just below :)